src/Lime.Transport.SignalR/SignalRTransportListener.cs

```diff
@@ -0,0 +1,224 @@
+
+                await _webHost.StartAsync(cancellationToken).ConfigureAwait(false);
+            }
+            finally
+            {
+                _semaphore.Release();
+            }
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="InvalidOperationException">Thrown when the listener is not started.</exception>
+        public async Task<ITransport> AcceptTransportAsync(CancellationToken cancellationToken)
+        {
+            if (!IsStarted)
+            {
+                throw new InvalidOperationException("The listener is not started");
+            }
+
+            return await _transportChannel.Reader.ReadAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="InvalidOperationException">Thrown when the listener is not started.</exception>
+        public async Task StopAsync(CancellationToken cancellationToken)
+        {
+            await _semaphore.WaitAsync().ConfigureAwait(false);
+            try
+            {
+                if (!IsStarted)
+                {
+                    throw new InvalidOperationException("The listener is not started");
+                }
+
+                _transportChannel.Writer.Complete();
+
+                await _webHost.StopAsync(cancellationToken).ConfigureAwait(false);
+            }
+            finally
+            {
+                _semaphore.Release();
+            }
+        }
+
+        private IHost BuildWebHost(
+            IEnvelopeSerializer envelopeSerializer,
+            X509Certificate2 tlsCertificate,
+            int bufferSize,
+            ITraceWriter traceWriter,
+            HttpProtocols httpProtocols,
+            SslProtocols sslProtocols,
+            Func<X509Certificate2, X509Chain, SslPolicyErrors, bool> clientCertificateValidationCallback,
+            TimeSpan? keepAliveInterval)
+        {
+            HttpConnectionDispatcherOptions httpConnectionDispatcherOptions = null;
+            HubOptions hubOptions = null;
+
+            return Host.CreateDefaultBuilder()
+                    .ConfigureWebHostDefaults(webBuilder =>
+                    {
+                        webBuilder.UseKestrel(serverOptions =>
+                        {
+                            foreach (var listenerUri in ListenerUris)
+                            {
+                                if (!IPAddress.TryParse(listenerUri.Host, out var ipAddress))
+                                {
+                                    ipAddress = IPAddress.Any;
+                                }
+
+                                var endPoint = new IPEndPoint(ipAddress, listenerUri.Port);
+                                serverOptions.Listen(endPoint, listenOptions =>
+                                {
+                                    listenOptions.Protocols = httpProtocols;
+
+                                    if (listenerUri.Scheme == URI_SCHEME_HTTPS)
```

Comment:
Use `Uri.UriSchemeHttps`

```suggestion
                                    if (listenerUri.Scheme == Uri.UriSchemeHttps)
```